packages/semcom-components/lib/components/sem-com-register.machine.ts

```diff
@@ -0,0 +1,456 @@
+  constructor(public dropDown: string, public freeInput: string) {}
+
+}
+
+export class HasPermissionEvent implements EventObject {
+
+  public type: SemComRegisterEvents.HAS_PERMISSION = SemComRegisterEvents.HAS_PERMISSION;
+  constructor(public url: string) {}
+
+}
+
+export class NoPermissionEvent implements EventObject {
+
+  public type: SemComRegisterEvents.NO_PERMISSION = SemComRegisterEvents.NO_PERMISSION;
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor() {}
+
+}
+
+export class BackToStoreSelectionEvent implements EventObject {
+
+  public type: SemComRegisterEvents.BACK_TO_STORE_SELECTION = SemComRegisterEvents.BACK_TO_STORE_SELECTION;
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor() {}
+
+}
+
+export class BackToUploadFormEvent implements EventObject {
+
+  public type: SemComRegisterEvents.BACK_TO_UPLOAD_FORM = SemComRegisterEvents.BACK_TO_UPLOAD_FORM;
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor() {}
+
+}
+
+export class UploadFormSubmittedEvent implements EventObject {
+
+  public type: SemComRegisterEvents.UPLOAD_FORM_SUBMITTED = SemComRegisterEvents.UPLOAD_FORM_SUBMITTED;
+  constructor (public uploadFormContext: UploadFormContext) {}
+
+}
+
+export class DataSavedEvent implements EventObject {
+
+  public type: SemComRegisterEvents.DATA_SAVED = SemComRegisterEvents.DATA_SAVED;
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor() {}
+
+}
+
+export class DataNotSavedEvent implements EventObject {
+
+  public type: SemComRegisterEvents.DATA_NOT_SAVED = SemComRegisterEvents.DATA_NOT_SAVED;
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor() {}
+
+}
+
+export type SemComRegisterEvent =
+| AuthenticatedEvent
+| StoreSelectedEvent
+| HasPermissionEvent
+| NoPermissionEvent
+| BackToStoreSelectionEvent
+| BackToUploadFormEvent
+| UploadFormSubmittedEvent
+| DataSavedEvent
+| DataNotSavedEvent;
+
+/* MACHINE */
+
+export const semComRegisterMachine: MachineConfig<
+SemComRegisterContext,
+SemComRegisterStateSchema,
+SemComRegisterEvent> = {
+
+  initial: SemComRegisterStates.AUTHENTICATING,
+  context: {},
+  states: {
+
+    [SemComRegisterStates.AUTHENTICATING]: {
+      on: {
+        [SemComRegisterEvents.AUTHENTICATED]: {
+          target: SemComRegisterStates.STORE_SELECTION,
+        },
+      },
+    },
+
+    [SemComRegisterStates.STORE_SELECTION]: {
+      invoke: {
+        id: FormActors.FORM_MACHINE,
+        src: formMachine<{ dropDown: string; freeInput: string }> ((context) => validateStoreSelectionForm(context)),
+        data: { data: { dropDown: 'empty', freeInput: '' }, original: { dropDown: 'empty', freeInput: '' } },
+        onDone: { actions: send(
+          (_, event) => new StoreSelectedEvent(event.data.data.dropDown, event.data.data.freeInput)
+        ) },
```

Comment:
I know this is how it is done in NDE, but I'd prefer the Form Machines to be included in their Lit components, and when done/submitted, alert this machine by way of an event.

Reply:
The form machine is alerted by an event in the `sem-com-store-selection.component.ts` on line 115.

Reply:
I don't know how I would include the form machine in the Lit component itself, so would need some help with that.